x-pack/legacy/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -0,0 +1,460 @@
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/:jobId Get anomaly detection data by id
+   * @apiName GetAnomalyDetectorsById
+   * @apiDescription Returns the anomaly detection job.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobId } = request.params;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.jobs', { jobId });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/_stats Get anomaly detection stats
+   * @apiName GetAnomalyDetectorsStats
+   * @apiDescription Returns anomaly detection jobs statistics.
+   *
+   * @apiSuccess {Number} count
+   * @apiSuccess {Object[]} jobs
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/_stats',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.jobStats');
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/:jobId/_stats Get stats for requested anomaly detection job
+   * @apiName GetAnomalyDetectorsStatsById
+   * @apiDescription Returns anomaly detection job statistics.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/_stats',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobId } = request.params;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.jobStats', { jobId });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {put} /api/ml/anomaly_detectors/:jobId Instantiate an anomaly detection job
+   * @apiName CreateAnomalyDetectors
+   * @apiDescription This API creates an anomaly detection job that performs an analysis
```

Comment:
Simplify this to just `Creates an anomaly detection job.`

Reply:
Updated b70059756cf4d0eb6a125b0ba7e891c69daa622d